test/test_weak.py

```diff
@@ -562,5 +570,340 @@ def test_pop(self):
         self.assertRaises(KeyError, d.pop, k)
 
 
+# Adapted from cpython/Lib/test/mapping_tests.py
+class WeakKeyDictionaryScriptObjectTestCase(TestCase):
+    def _reference(self):
+        self.__ref = {
+            torch.classes._TorchScriptTesting._Foo(1, 2): 1,
+            torch.classes._TorchScriptTesting._Foo(2, 3): 2,
+            torch.classes._TorchScriptTesting._Foo(3, 4): 3,
+        }
+        return self.__ref.copy()
+
+    def _empty_mapping(self):
+        """Return an empty mapping object"""
+        return WeakIdKeyDictionary(ref_type=WeakScriptObjectRef)
+
+    def _full_mapping(self, data):
+        """Return a mapping object with the value contained in data
+        dictionary"""
+        x = self._empty_mapping()
+        for key, value in data.items():
+            x[key] = value
+        return x
+
+    def __init__(self, *args, **kw):
+        if IS_MACOS:
+            raise unittest.SkipTest("non-portable load_library call used in test")
+
+        if IS_SANDCASTLE or IS_FBCODE:
+            torch.ops.load_library(
+                "//caffe2/test/cpp/jit:test_custom_class_registrations"
+            )
+        else:
+            lib_file_path = find_library_location("libtorchbind_test.so")
+            if IS_WINDOWS:
+                lib_file_path = find_library_location("torchbind_test.dll")
+            torch.ops.load_library(str(lib_file_path))
+
+        unittest.TestCase.__init__(self, *args, **kw)
+        self.reference = self._reference().copy()
```

Comment:
This will be shared by all tests. Shouldn't this be in setUp() ?

Reply:
this is part of the code copied from cpython; I'd prefer not to change unless necessary lol